### ISSUE TITLE
All Languages: provide a listing of code generators with their custom settings

### ISSUE THREAD

**Issue by devigned** [Azure/autorest#753]
Currently:

```
$ mono binaries/net45/AutoRest.exe
Microsoft (R) AutoRest 0.15.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Usage: AutoRest.exe -Input <value> [-Namespace <value>] [-OutputDirectory <value>] [-CodeGenerator <value>] [-Modeler <value>] [-ClientName <value>] [-PayloadFlatteningThreshold <value>] [-Header <value>] [-AddCredentials <value>] [-OutputFileName <value>] [-PackageName <value>] [-PackageVersion <value>]

Parameters:
-Input: The location of the input specification. Aliases: -i, -input
-Namespace: The namespace to use for generated code. Aliases: -n
-OutputDirectory: The location for generated files. If not specified, uses "Generated" as the default. Aliases: -o, -output
-CodeGenerator: The code generator language. If not specified, defaults to CSharp. Aliases: -g
-Modeler: The Modeler to use on the input. If not specified, defaults to Swagger. Aliases: -m
-ClientName: Name to use for the generated client type. By default, uses the value of the 'Title' field from the Swagger input. Aliases: -name
-PayloadFlatteningThreshold: The maximum number of properties in the request body. If the number of properties in the request body is less than or equal to this value, these properties will be represented as method arguments. Aliases: -ft
-Header: Text to include as a header comment in generated files. Use NONE to suppress the default header. Aliases: -header
-AddCredentials: If true, the generated client includes a ServiceClientCredentials property and constructor parameter. Authentication behaviors are implemented by extending the ServiceClientCredentials type.
-OutputFileName: If set, will cause generated code to be output to a single file. Not supported by all code generators.
-PackageName: Package name of then generated code package. Should be then names wanted for the package in then package manager. Aliases: -pn
-PackageVersion: Package version of then generated code package. Should be then version wanted for the package in then package manager. Aliases: -pv


Examples:
Generate C# client in MyNamespace from swagger.json input:
  >AutoRest.exe -Namespace MyNamespace -Input swagger.json

Generate C# client in MyNamespace including custom header from swagger.json input:
  >AutoRest.exe -Namespace MyNamespace -Header "Copyright Contoso Ltd" -Input swagger.json

Generate C# client with a credentials property in MyNamespace from swagger.json input:
  >AutoRest.exe -AddCredentials true -Namespace MyNamespace -CodeGenerator CSharp -Modeler Swagger -Input swagger.json
```

What would be nice:

```
$ mono binaries/net45/AutoRest.exe
Microsoft (R) AutoRest 0.15.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Usage: AutoRest.exe -Input <value> [-Namespace <value>] [-OutputDirectory <value>] [-CodeGenerator <value>] [-Modeler <value>] [-ClientName <value>] [-PayloadFlatteningThreshold <value>] [-Header <value>] [-AddCredentials <value>] [-OutputFileName <value>] [-PackageName <value>] [-PackageVersion <value>]

Parameters:
-Input: The location of the input specification. Aliases: -i, -input
-Namespace: The namespace to use for generated code. Aliases: -n
....
-PackageVersion: Package version of then generated code package. Should be then version wanted for the package in then package manager. Aliases: -pv

Generators:
- CSharp: for generating CSharp
  Parameters:
    -SomethingCool: The awesome thing that only applies to CSharp
- Azure.CSharp: for generating CSharp with an Azure opinion 
- Java
...

Examples:
...
```

**Comment by devigned** [comment]
@stankovski thoughts?

**Comment by stankovski** [comment]
I like it :)

**Comment by stankovski** [comment]
Reopening as we need a list of code generator specific settings like "useDateTimeOffset" documented.

**Comment by fearthecowboy** [comment]
Fixed by #1569
